content/docs/reference/design-docs/train-sim-v3/_index.en.md

```diff
@@ -0,0 +1,514 @@
+
+InitTrainState ---> ScheduleController
+
+TrainSim --> SimResults
+
+TrainSim([train simulator])
+SimResults[simulation result curve]
+```
+
+
+### Driving instructions
+
+Driving instructions model what the train has to do, and under what conditions.
+Driving instructions are generated using domain constraints such as:
+
+- unsignaled line speed limits
+- permanent signaled speed limits
+- temporary speed limits
+- dynamic signaling:
+  - block / moving block
+  - dynamically signaled speed restrictions
+- neutral zones
+- stops
+- margins
+
+
+There are two types of driving instructions:
+
+- **Abstract driving instructions** model the high-level, rolling stock independent
+range of acceptable behavior: reach 30km/h at this location
+- **Concrete driving instructions** model the specific range of acceptable behavior
+for a specific rolling stock, using limit curves: don't go faster than this curve
+
+```mermaid
+flowchart TD
+Constraint[constraint]
+AbstractDrivingInstruction[abstract driving instruction]
+ConcreteDrivingInstruction[concrete driving instruction]
+RollingStockIntegrator[rolling stock integrator]
+Compiler([compiler])
+
+Constraint -- generates one or more --> AbstractDrivingInstruction
+AbstractDrivingInstruction --> Compiler
+RollingStockIntegrator --> Compiler
+Compiler --> ConcreteDrivingInstruction
+```
+
+
+#### Interpreting driving instructions
+
+During the simulation, driving instructions are partitionned into 4 sets:
+
+- `PENDING` instructions may apply at some point in the future
+- `RECEIVED` instructions aren't enforced yet, but will be unless overridden
+- `ENFORCED` instructions influence train behavior
+- `DISABLED` instructions don't ever have to be considered anymore. There are multiple ways instructions can be disabled:
+  - `SKIPPED` instructions were not received
+  - `RETIRED` instructions expired by themselves
+  - `OVERRIDEN` instructions were removed by another instruction
+
+```mermaid
+flowchart TD
+
+subgraph disabled
+    skipped
+    retired
+    overriden
+end
+
+subgraph active
+    received
+    enforced
+end
+
+pending --> received
+pending --> skipped
+received --> enforced
+received --> overriden
+enforced --> retired
+enforced --> overriden
+```
+
+These sets evolve as follows:
+
+- when an integration steps overlaps a `PENDING` instruction's received condition, it is `RECEIVED` and becomes a candidate to execution
+  - existing instructions may be `OVERRIDEN` due to an `override_on_received` operation
+- if an instruction cannot ever be received at any future simulation state, it transitions to the `SKIPPED` state
+- when simulation state exceeds an instruction's enforcement position, it becomes `ENFORCED`. Only enforced instructions influence train behavior.
+  - existing instructions may be `OVERRIDEN` due to an `override_on_enforced` operation
+- when simulation state exceeds an instruction's retirement position, it becomes `RETIRED`
+
+
+#### Overrides
```

Comment:
This is quite a rich algorithm maybe we'd need a set of examples to explain why each detail of the automaton is needed, and why they're sufficient
Maybe we could do that when it's implemented and we have a better understanding of it.